### PR TITLE
Fix/Comment/Remove tests depending on status

### PR DESCRIPTION
**Pull request by alexr-bq**
This PR does 3 things:
- Fixes some tests, mostly changes in syntax from Postgres to Vertica
- Comments out some tests for not-yet-supported functionality
- Removes tests related to Postgres-specific functionality that Vertica will not support

### CODE REVIEW

File: packages/vertica-nodejs/script/create-test-tables.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ con.connect((err) => {
       console.log('Created table person')
       console.log('Filling it with people')
 
-      let q = people.map((person) => `INSERT INTO person (name, age) VALUES ('${person.name}', ${person.age})`).join(';')
+      let q = people.map((person) => `INSERT INTO person (name, age) VALUES ('${person.name}', ${person.age})`).join(';') + ";COMMIT;"
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/script/create-test-tables.js:71]: Will we be removing this file entirely in favour of the local temp tables Danny suggested?

File: packages/vertica-nodejs/test/integration/client/connection-parameter-tests.js [hunk 2]
```diff
@@ -3,13 +3,15 @@ const helper = require('../test-helper')
 const suite = new helper.Suite()
 const { Client } = helper.vertica
 
+/*
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/connection-parameter-tests.js:7]: Should leave a note here about why this is disabled

File: packages/vertica-nodejs/test/integration/client/query-error-handling-prepared-statement-tests.js [hunk 3]
```diff
@@ -42,84 +42,4 @@ suite.test('client end during query execution of prepared statement', function (
...
-              assert.emits(client2, 'end')
-            })
-          )
-        })
-      )
-    })
-  )
-}
-
-suite.test('query killed during query execution of prepared statement', function (done) {
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/query-error-handling-prepared-statement-tests.js:85]: Why are we removing this test?

File: packages/vertica-nodejs/test/integration/client/row-description-on-results-tests.js [hunk 4]
```diff
@@ -21,8 +21,9 @@ test('row descriptions on result object', function () {
   var client = new Client(conInfo)
   client.connect(
     assert.success(function () {
+      console.log("111")
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/row-description-on-results-tests.js]: What are these console.logs for?

File: packages/vertica-nodejs/test/integration/gh-issues/507-tests.js [hunk 5]
```diff
@@ -6,9 +6,10 @@ new helper.Suite().test('parsing array results', function (cb) {
   const pool = new vertica.Pool()
   pool.connect(
     assert.success(function (client, done) {
-      client.query('CREATE TEMP TABLE test_table(bar integer, "baz\'s" integer)')
+      client.query('CREATE LOCAL TEMP TABLE test_table(bar integer, "baz\'s" integer)')
       client.query('INSERT INTO test_table(bar, "baz\'s") VALUES(1, 1), (2, 2)')
       client.query('SELECT * FROM test_table', function (err, res) {
+        console.log(res.rows)
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/gh-issues/507-tests.js]: Why is there a console.log here?

File: packages/vertica-nodejs/test/integration/connection-pool/error-tests.js [hunk 6]
```diff
@@ -9,134 +9,6 @@ suite.test('connecting to invalid port', (cb) => {
   pool.connect().catch((e) => cb())
 })
 
-suite.test('errors emitted on checked-out clients', (cb) => {
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/connection-pool/error-tests.js:12]: Why were these tests deleted?

File: packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js [hunk 7]
```diff
@@ -3,11 +3,12 @@ var helper = require('./test-helper')
...
 function testTypeParser(client, expectedResult, done) {
   var boolValue = true
-  client.query('CREATE TEMP TABLE parserOverrideTest(id bool)')
-  client.query('INSERT INTO parserOverrideTest(id) VALUES ($1)', [boolValue])
+  client.query('CREATE LOCAL TEMP TABLE parserOverrideTest(id bool)')
+  client.query('INSERT INTO parserOverrideTest(id) VALUES (?)', [boolValue])
   client.query(
     'SELECT * FROM parserOverrideTest',
     assert.success(function (result) {
+      console.log(result)
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js]: Why was this console.log added?

File: packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js [hunk 8]
```diff
@@ -19,7 +20,7 @@ pool.connect(
   assert.success(function (client1, done1) {
     pool.connect(
       assert.success(function (client2, done2) {
-        var boolTypeOID = 16
+        var boolTypeOID = 5
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js]: Can we use the OIDs from the VerticaType enum instead of just a numeric literal?

File: packages/vertica-nodejs/test/integration/client/transaction-tests.js [hunk 9]
```diff
@@ -18,6 +18,7 @@ client.connect(
         getZed,
         assert.calls(function (err, result) {
           assert(!err)
+          console.log(result.rows)
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/transaction-tests.js]: Why was this console.log added?

File: packages/vertica-nodejs/test/integration/client/result-metadata-tests.js [hunk 10]
```diff
@@ -7,39 +7,32 @@ new helper.Suite().test('should return insert metadata', function () {
...
+                  assert.equal(result.command, 'SELECT')
+                  done()
+                  process.nextTick(pool.end.bind(pool))
                 })
               )
             })
           )
         })
-      )
+        )
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/result-metadata-tests.js:35]: nit: indentation is off

File: packages/vertica-nodejs/test/integration/client/result-metadata-tests.js [hunk 11]
```diff
@@ -7,39 +7,32 @@ new helper.Suite().test('should return insert metadata', function () {
...
     assert.calls(function (err, client, done) {
       assert(!err)
+      client.query(
+        'CREATE TEMP TABLE zugzug(name varchar(10))',
+        assert.calls(function (err, result) {
+          assert(!err)
+          assert.equal(result.oid, null)
+          assert.equal(result.command, 'CREATE')
 
-      helper.versionGTE(
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/result-metadata-tests.js:11]: Why is this removed? I'm not sure what this was originally trying to do.

File: packages/vertica-nodejs/test/integration/client/query-error-handling-tests.js [hunk 12]
```diff
@@ -4,119 +4,38 @@ var util = require('util')
 var Query = helper.vertica.Query
 var DatabaseError = helper.vertica.DatabaseError
 
-test('error during query execution', function () {
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/query-error-handling-tests.js:7]: Why was this test deleted?

**Review by jonathanl-bq — COMMENTED** [review]
Left some questions and a couple of minor suggestions.

File: packages/vertica-nodejs/script/create-test-tables.js (hunk 1 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/script/create-test-tables.js:71]: Should be an eventual goal but would need to do it for a lot of tests so imo should be lower priority.

File: packages/vertica-nodejs/test/integration/client/connection-parameter-tests.js (hunk 2 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/connection-parameter-tests.js:7]: Done

File: packages/vertica-nodejs/test/integration/client/query-error-handling-prepared-statement-tests.js (hunk 3 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/query-error-handling-prepared-statement-tests.js:85]: Postgres-specific functionality to kill a specific query.

File: packages/vertica-nodejs/test/integration/client/row-description-on-results-tests.js (hunk 4 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/row-description-on-results-tests.js]: removing

File: packages/vertica-nodejs/test/integration/gh-issues/507-tests.js (hunk 5 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/gh-issues/507-tests.js]: removing

File: packages/vertica-nodejs/test/integration/client/transaction-tests.js (hunk 9 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/transaction-tests.js]: removing

File: packages/vertica-nodejs/test/integration/client/result-metadata-tests.js (hunk 11 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/result-metadata-tests.js:11]: Anywhere you see the versionGTE() call it's testing some split in functionality between two different version of postgres. Does not apply to vertica.

File: packages/vertica-nodejs/test/integration/client/query-error-handling-tests.js (hunk 12 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/query-error-handling-tests.js:7]: More postgres-specific functionality.

File: packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js (hunk 8 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js]: done

File: packages/vertica-nodejs/script/create-test-tables.js (hunk 1 above)
Comment (williammatherjones) [inline packages/vertica-nodejs/script/create-test-tables.js:71]: Is there any way we can build calls to commit into a framework?  Seems like it runs the risk of being less obvious vs more reliable and consistent.

Comment (DMickens) [reply packages/vertica-nodejs/script/create-test-tables.js:71]: An alternative is just having autocommit enabled by default or set for tests when I add that property

File: packages/vertica-nodejs/test/integration/client/configuration-tests.js [hunk 13]
```diff
@@ -16,27 +16,27 @@ suite.test('default values are used in new clients', function () {
...
     parseInputDatesAsUTC: false,
   })
 
   var client = new vertica.Client()
   assert.same(client, {
-    user: process.env.USER,
-    database: process.env.USER,
     password: null,
-    port: 5432,
+    port: 5433,
```
Comment (williammatherjones) [inline packages/vertica-nodejs/test/integration/client/configuration-tests.js]: Isn't it bad to have our tests hard code the port?   I would think we would want to depend on the V_* environment variables.

File: packages/vertica-nodejs/test/integration/client/api-tests.js [hunk 14]
```diff
@@ -9,19 +9,24 @@ suite.test('null and undefined are both inserted as NULL', function (done) {
   pool.connect(
     assert.calls(function (err, client, release) {
       assert(!err)
-      client.query('CREATE TEMP TABLE my_nulls(a varchar(1), b varchar(1), c integer, d integer, e date, f date)')
-      client.query('INSERT INTO my_nulls(a,b,c,d,e,f) VALUES ($1,$2,$3,$4,$5,$6)', [
+      client.query('DROP TABLE IF EXISTS my_nulls; CREATE TEMP TABLE my_nulls(a varchar(1), b varchar(1), c integer, d integer, e date, f date)')
```
Comment (DMickens) [inline packages/vertica-nodejs/test/integration/client/api-tests.js]: Didn't we decide to just use local temp tables?

File: packages/vertica-nodejs/test/integration/client/query-error-handling-prepared-statement-tests.js [hunk 15]
```diff
@@ -42,84 +42,4 @@ suite.test('client end during query execution of prepared statement', function (
...
-
-      query1.on('end', function (err) {
-        assert.fail('Prepared statement when executed should not return before being killed')
-      })
-
-      killIdleQuery(sleepQuery, done)
-    })
-  )
-})
+})
```
Comment (DMickens) [inline packages/vertica-nodejs/test/integration/client/query-error-handling-prepared-statement-tests.js]: nit: add back newline

File: packages/vertica-nodejs/test/integration/client/row-description-on-results-tests.js [hunk 16]
```diff
@@ -47,4 +47,4 @@ test('row description on no rows', function () {
       )
     })
   )
-})
+})
```
Comment (DMickens) [inline packages/vertica-nodejs/test/integration/client/row-description-on-results-tests.js]: nit: add back newline

File: packages/vertica-nodejs/test/integration/client/configuration-tests.js (hunk 13 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/configuration-tests.js]: This test is meant to test what the default values are, but you are right that it doesn't play nice with our environment variables. Not sure what to do with this one.

File: packages/vertica-nodejs/script/create-test-tables.js (hunk 1 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/script/create-test-tables.js:71]: It would be a good feature to have. @DMickens we do not currently have this property, correct?

File: packages/vertica-nodejs/test/integration/client/api-tests.js (hunk 14 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/api-tests.js]: Yeah, added this before that, will change

File: packages/vertica-nodejs/test/integration/client/configuration-tests.js (hunk 13 above)
Comment (DMickens) [reply packages/vertica-nodejs/test/integration/client/configuration-tests.js]: The problem is that if environment variables are set then that client created with no config object is going to use the environment variables instead of the defaults (which may or may not be the same, probably will be most of the time). However starting on line 15 we just asserted that the all the defaults are the same as what was hard coded, is there any real benefit from having this client comparison assert that we don't already essentially get from asserting the defaults are what we expect they are? If not I would say just remove that part of the test

File: packages/vertica-nodejs/script/create-test-tables.js (hunk 1 above)
Comment (DMickens) [reply packages/vertica-nodejs/script/create-test-tables.js:71]: I have it mostly ready to go in it's own branch, should probably just give it a test, but we've decided to curb any new features until we get all of these passing tests. I can make a note to remove any soon-to-be unnecessary commits whenever the autocommit PR makes it through.

File: packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js (hunk 7 above)
Comment (DMickens) [reply packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js]: +1. There are other tests as well on master which also seem to have some stray logged results/errors. Once we get everything passing we can make a quick pass through and remove console output

**Review by DMickens — APPROVED** [review]
I have no further concerns other than what's still open in conversations. Once existing comments have been addressed I think this is good to go.

File: packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js (hunk 7 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js]: removing

File: packages/vertica-nodejs/test/integration/client/configuration-tests.js (hunk 13 above)
Comment (alexr-bq) [reply packages/vertica-nodejs/test/integration/client/configuration-tests.js]: Good point, removing.
